lib/puppet-lint/plugins/top_scope_facts/top_scope_facts.rb

```diff
@@ -0,0 +1,38 @@
+# Public: A puppet-lint plugin that will check for the use of top scope facts.
+# For example, the fact `$facts['os']['name']` should be used over
+# `$::operatingsystem`.
```

Comment:
I still think this is a bad example. `$::operatingsystem` will be converted by this plugin to `$facts['operatingsystem']` and then puppet-lint-legacy_facts-check will convert it into `$facts['os']['name']`

```suggestion
# For example, the fact `$facts['kernel'] should be used over
# `$::kernel`.
```

Reply:
Nice one. Updated.